### ISSUE TITLE
Use QR code for bitcoin donation

### ISSUE THREAD

**Issue by tursiops33** [diaspora/diaspora#7468]
Hi everyone 
I feel like it would be more visible and it would have a bigger impact if the bitcoin address field would be replaced or even just supplemented by a QR code. Most users of bitcoin use their phone for payments and its always a nice commodity to be able to just scan the code instead of having to enter manually the bitcoin wallet address.

What do you think?
Any comments appreciated.
Thanks.

**Comment by oneturkmen** [comment]
Hi there! I want to take this issue, but still researching how to do it best.

Do you guys think (suggestions/comments are more than welcome) if this should be done on the back-end or front-end side? I think this can be done with front-end side javascript using qrcode.js, but what is better?

**Comment by svbergerem** [comment]
We could keep the bitcoin address and add a link below that opens a modal with the QR code. This could be rendered using JavaScript.

**Comment by oneturkmen** [comment]
Two follow-up questions:

1. What if add bitcoin qr code straight below the address without providing link to qr?


2. Found a way to generate QR code using Google Charts API. 
Example: `https://chart.googleapis.com/chart?chs=250x250&cht=qr&chl={BITCOIN-ADDRESS-HERE}`

However, is it secure to use their API? And is the bitcoin address field dynamic (i.e. changes through time)?

**Comment by SuperTux88** [comment]
> Found a way to generate QR code using Google Charts API.

Please don't use Google Chart API or any other external API, use something that is locally on the pod.

**Comment by svbergerem** [comment]
> What if add bitcoin qr code straight below the address without providing link to qr?

I don't think that we should generate the QR code every time we render the page. That's why I suggested to only render it when the user clicks a link.

**Comment by tursiops33** [comment]
Hi, I found the following project on github that might help to generate the QR code: https://github.com/AliasIO/BitcoinQRCode.org 
Hope it helps.
Thanks!

**Comment by oneturkmen** [comment]
I found a way to generate QR "in-place" (i.e. using jquery) within the modal. As of now, I am having some issues with installing Diaspora locally, thus, not able to fully test it. But it's coming! :soon:

**Comment by tursiops33** [comment]
Hi everyone. Any luck with this particular topic?
